### PR TITLE
feat: add telemetry for app studio plugin

### PR DESCRIPTION
**Pull request by nliu-ms**

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/appstudio/utils/telemetry.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+enum TelemetryPropertyValue {
+    UserError = "UserError",
+    SystemError = "SystemError",
+    UnhandledError = "UnhandledError"
+}
+
+export enum TelemetryEventName {
+    validateManifest = "validateManifest",
```
Comment (chagong) [inline packages/fx-core/src/plugins/resource/appstudio/utils/telemetry.ts:22]: to snake case style: "validate-manifest".

File: packages/fx-core/src/plugins/resource/appstudio/utils/telemetry.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+export enum TelemetryPropertyKey {
+    component = "component",
+    errorType = "error-type",
+    errorCode = "error-code",
+    errorMessage = "error-message",
+    validationResult = "validation-result"
+}
+
+enum TelemetryPropertyValue {
+    UserError = "UserError",
```
Comment (chagong) [inline packages/fx-core/src/plugins/resource/appstudio/utils/telemetry.ts:16]: "user"

File: packages/fx-core/src/plugins/resource/appstudio/utils/telemetry.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    SystemError = "SystemError",
```
Comment (chagong) [inline packages/fx-core/src/plugins/resource/appstudio/utils/telemetry.ts:17]: system

File: packages/fx-core/src/plugins/resource/appstudio/utils/telemetry.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    UnhandledError = "UnhandledError"
```
Comment (chagong) [inline packages/fx-core/src/plugins/resource/appstudio/utils/telemetry.ts:18]: No 3rd type of error. Unhandled error should be system error.

File: packages/fx-core/src/plugins/resource/appstudio/utils/telemetry.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+export enum TelemetryEventName {
+    validateManifest = "validateManifest",
+    buildTeamsPackage = "buildTeamsPackage",
```
Comment (chagong) [inline packages/fx-core/src/plugins/resource/appstudio/utils/telemetry.ts:23]: snake case style. telemetry pipeline will make every thing lower case. And use "build-start" and "build" pair to address start and exit of operation.

**Review by chagong — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**